### PR TITLE
Add list of task and reward points

### PR DESCRIPTION
**Pull request by DinMon**
- Display a list of task to complete together with its reward points.
- Set colour of task card by the reward points, the higher the point, the more reddish the colour will be.
- Sort task in descending order by the reward points

Time: 2 h 10 min

### CODE REVIEW

**Comment by DinMon** [comment]
- Refactored TaskCard and TaskGrid components to use Task and Task holder to separate logic in domain entities.

Time: 2 h 38 min

File: react-app/src/components/TaskGrid.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import React, { useEffect, useRef, useState } from 'react'
+import { createTask } from '../domain/Task'
+import { createTaskHolder, DESC } from '../domain/TaskHolder'
+import TaskCard from './TaskCard';
+
+function TaskGrid({}) {
+    const taskHolder = useRef(null)
+    const [tasks, setTasks] = useState([])
```
Comment (VitalyKrenel) [inline react-app/src/components/TaskGrid.js:8]: JFYI: I'm the proponent of always using semicolons, so usually in the teams, I'm working with I advise you to stick with them and use them always (as ASI is not 100% reliable).

image

In my experience, most of the projects were always finishing ending with a semicolon, but It is debatable for sure, so I'll leave this up to.

File: react-app/src/components/TaskGrid.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            .add(createTask({ title: 'Put away old toys to white boxes on the balconey', rewardPoints: 100 }))
+            .add(createTask({ title: 'Wash the dishes', rewardPoints: 75 }))
+            .add(createTask({ title: 'Make your bed in the morning', rewardPoints: 20 }))
+            .add(createTask({ title: 'Fix the pillowcase', rewardPoints: 50 }))
+            .add(createTask({ title: 'Water the plants', rewardPoints: 80 }))
+            .sortBy(DESC)
+
+        const { tasks } = taskHolder.current
+        setTasks(tasks)
+    }, [])
```
Comment (VitalyKrenel) [inline react-app/src/components/TaskGrid.js:22]: I think we discussed this over the meeting, but I'll summarize: try to use a more simple approach, to make sure that your code is simple enough to be maintained and updated without much effort in the future :wink:

**Review by VitalyKrenel — APPROVED** [review]
@DinMon 

Let's see your work in the second PR, as we discussed quite a few things so the codebase will be quite different from the current implementation.

Good work, keep it up! Waiting for the second task to be done!

**Pull request merged**